### PR TITLE
bumps sev crate to 3.1.1

### PR DESCRIPTION
**Pull request by wobito**
This PR bumps the SEV crate for support in https://github.com/confidential-containers/trustee/pull/385
and handles the new return structure from SEV 3.1.1

### CODE REVIEW

File: attestation-agent/attester/src/snp/mod.rs [hunk 1]
```diff
@@ -44,6 +44,11 @@ impl Attester for SnpAttester {
             .get_ext_report(None, Some(data), Some(0))
             .context("Failed to get attestation report")?;
 
+        let certs = match certs {
+            Some(c) => c,
+            None => bail!("Certificate chain was not provided"),
+        };
```
Comment (fitzthum) [inline attestation-agent/attester/src/snp/mod.rs:50]: I think we are going to add support in Trustee for the verifier to pull the cert chain if needed, so I would prefer not to have a fatal error here.

**Review by fitzthum — COMMENTED** [review]
Looks good. One note and it looks like you might have a conflict.

I am slightly worried about picking up some random changes with this jump, but let's see.

**Comment by wobito** [comment]
Thanks for the comments will pick this up tomorrow and rectify the conflict. 

We have tested this in aws with the verifier changes in my other pr mentioned above and bare metal using the kbs-client.

File: attestation-agent/attester/src/snp/mod.rs (hunk 1 above)
Comment (mkulke) [inline attestation-agent/attester/src/snp/mod.rs:50]: apart from the @fitzthum comment, this pattern matching can be rewritten to an if let

```suggestion
        let Some(certs) = certs else {
             bail!("Certificate chain was not provided");
        }
```

Comment (wobito) [reply attestation-agent/attester/src/snp/mod.rs]: I've removed the bail and just unwrapped the value.

Comment (fitzthum) [reply attestation-agent/attester/src/snp/mod.rs]: Unwrapping it is still fatal. If there is no cert chain we probably want to pass an empty Vec. You could do `cert_chain: certs.unwrap_or(vec![])` or we could change the SnpEvidence struct to have `cert_chain: Option<Vec<CertTableEntry>>`

Comment (fitzthum) [reply attestation-agent/attester/src/snp/mod.rs]: Looks like the SEV crate returns `Option<Vec<CertTableEntry>>` so you might just change the struct and use that value directly.

**Review by mkulke — APPROVED** [review]
lgtm.

I think we probably want to start versioning the evidence structs somewhere. Maybe not explicitly, but at the moment it's somewhat intransparent which pair of attester and verifier will work.

I assume sev's v3 `AttestationReport` struct will also be different from the one in v1.

the trustee repo references guest-components in the kbs/tools/client crate, it might be a good idea to bump it there together with the verifier PR, so we have a least a working pair documented.

**Review by fitzthum — APPROVED** [review]
LGTM. Let's wait to merge this until the verifier PR is a bit closer since we'll need to fixup the struct there. ofc this will still have to go first.

**Comment by wobito** [comment]
Thanks everyone!! happy to contribute help!
Working on the verifier pr at the moment too.

**Pull request merged**